common/reviews/api/rush-lib.api.md

```diff
@@ -1068,18 +1066,29 @@ export class RushConfiguration {
     readonly experimentsConfiguration: ExperimentsConfiguration;
     findProjectByShorthandName(shorthandProjectName: string): RushConfigurationProject | undefined;
     findProjectByTempName(tempProjectName: string): RushConfigurationProject | undefined;
-    getCommittedShrinkwrapFilename(variant?: string | undefined): string;
-    getCommonVersions(variant?: string | undefined): CommonVersionsConfiguration;
-    getCommonVersionsFilePath(variant?: string | undefined): string;
+    getCommittedShrinkwrapFilename({ subspaceName, variant }: {
+        subspaceName?: string | undefined;
+        variant?: string | undefined;
+    }): string;
+    getCommittedSubspaceShrinkwrapFilename(subspaceName: string): string;
+    getCommonRushConfigFolder(subspaceName?: string | undefined): string;
+    getCommonTempFolder(subspaceName?: string | undefined): string;
+    getCommonVersions(subspaceName: string | undefined, variant?: string | undefined): CommonVersionsConfiguration;
+    getCommonVersionsFilePath(subspaceName: string | undefined, variant?: string | undefined): string;
     getImplicitlyPreferredVersions(variant?: string | undefined): Map<string, string>;
-    getPnpmfilePath(variant?: string | undefined): string;
+    getPackageManagerToolFilename(subspaceName?: string | undefined): string;
+    getPnpmfilePath(subspaceName: string | undefined, variant?: string | undefined): string;
     getProjectByName(projectName: string): RushConfigurationProject | undefined;
     // @beta (undocumented)
     getProjectLookupForRoot(rootPath: string): LookupByPath<RushConfigurationProject>;
-    getRepoState(variant?: string | undefined): RepoStateFile;
-    getRepoStateFilePath(variant?: string | undefined): string;
+    getProjectsSubspaceSet(projects: Set<RushConfigurationProject>): string[];
+    getProjectSubspace(project: RushConfigurationProject): string | undefined;
+    getRepoState(subspaceName: string | undefined, variant?: string | undefined): RepoStateFile;
+    getRepoStateFilePath(subspaceName: string | undefined, variant?: string | undefined): string;
```

Comment:
As part of introducing the subspaces, I think it's a good time to remove the [Installation Variants](https://rushjs.io/pages/advanced/installation_variants/) feature. The problem it solved can probably be handled using subspaces + `pnpm-sync`. And from discussions with the community, as far as we can tell nobody is actually using installation variants any more (because it only works for relatively small monorepos).

Ripping out that code may avoid some conflicts with the subspace APIs.  But maybe we should wait for your PR #4458 to get merged first to avoid a merge conflict.

Reply:
I've removed variants from many of the rush configuration apis, and I can remove the rest after this MR is merged